AirBench/Controllers/BenchController.cs

```diff
@@ -36,33 +40,80 @@ public ActionResult Create(CreateBench bench)
-            repository.InsertReview(myReview);
-            return RedirectToAction("Details");
+            CustomPrincipal currentUser = (CustomPrincipal)User;
+            if (ModelState.IsValid)
+            {
+                BenchRepository repository = new BenchRepository(context);
+                Review myReview = new Review(review.Rating, review.Description, currentUser.User.Id, review.BenchId);
+                repository.InsertReview(myReview);
+                return RedirectToAction("Index");
+            }
+            return View(review);
         }
 
         public ActionResult Index ()
         {
             BenchRepository repository = new BenchRepository(context);
             List<Bench> myBenches = repository.GetBenchList();
-            return View("Index", myBenches);
+            List<BenchList> myBenchList = new List<BenchList>();
+            foreach(var bench in myBenches)
+            {
+                var reviews = repository.GetReviewList(bench.Id);
+                BenchList currentBench = new BenchList();
+                currentBench.CreatorUserId = bench.CreatorUserId;
+                currentBench.Description = bench.Description;
+                currentBench.Latitude = bench.Latitude;
+                currentBench.Longitude = bench.Longitude;
+                currentBench.NumberOfSeats = bench.NumberOfSeats;
+                currentBench.NumberOfReviews = reviews.Count;
+                currentBench.Id = bench.Id;
+                decimal sum = 0;
+                foreach (var review in reviews)
+                {
+                    sum += (decimal)review.Rating;
+                }
+                if (reviews.Count != 0)
+                {
+                    currentBench.Rating = sum / reviews.Count;
+                }
+                else
+                {
+                    currentBench.Rating = 0;
+                }
+                myBenchList.Add(currentBench);
+            }
+
+            return View("Index", myBenchList);
         }
 
         public ActionResult Details (int id)
         {
             BenchRepository repository = new BenchRepository(context);
+            LoginRepository repo = new LoginRepository(context);
             Bench myBench = repository.GetById(id);
-            return View("Details", myBench);
+            User thisUser = repo.GetById(myBench.CreatorUserId);
+            BenchDetails myBenchDetails = new BenchDetails();
+            myBenchDetails.CreatorUserId = myBench.CreatorUserId;
+            myBenchDetails.CreatorUserName = thisUser.UserName;
+            myBenchDetails.Description = myBench.Description;
+            myBenchDetails.Latitude = myBench.Latitude;
+            myBenchDetails.Longitude = myBench.Longitude;
+            myBenchDetails.NumberOfSeats = myBench.NumberOfSeats;
+            myBenchDetails.reviews = repository.GetReviewList(id);
+            myBenchDetails.Id = myBench.Id;
```

Comment:
Again, I'd probably put all of this initialization code into a `BenchDetails` constructor.